### ISSUE TITLE
HardCodedArnProperties regex checks are too strict, and fail valid pseudo-parameters

### ISSUE THREAD

**Issue by cgledezma1101** [aws-cloudformation/cfn-lint#2543]
### CloudFormation Lint Version

0.72.5

### What operating system are you using?

Mac

### Describe the bug

`HardCodedArnProperties` uses the [following RegEx](https://github.com/aws-cloudformation/cfn-lint/blob/main/src/cfnlint/rules/resources/HardCodedArnProperties.py#L91) to check whether something is hardcoded:

```
 r"^\$\{\w+}|\$\{AWS::Partition}|$"
```

This is similar for region and account parameters. However, the `\$\{\w+}` element of this Regex is too restrictive, and fails validation for interpolations like those done for [ApiGateway's stage variables](https://docs.aws.amazon.com/apigateway/latest/developerguide/amazon-api-gateway-using-stage-variables.html)

For example, I'm trying to setup a header in one of my APIs to specify the format that an ARN should have. I do this by adding the following to my OpenAPI definition:

```
"id-format": "$arn{arn:${stageVariables.arnPartition}:service::${context.identity.accountId}:resource/${method.request.body.name}}"
```

and I define my stage variables in my API using:

```
MyApi:
  Type: AWS::Serverless:API
  Properies:
  ...
      Variables:
        arnPartition:
          Ref: "AWS::Partition"
```

This causes a validation failure:

```
I3042 ARN in Resource MyApi contains hardcoded Partition in ARN or incorrectly placed Pseudo Parameters
build/sam/template.yml
```

### Expected behavior

Validation for this sort of template should pass.

### Reproduction template

```yml
---
"AWSTemplateFormatVersion": "2010-09-09"
"Resources":
  "MyApi":
    "Properties":
      "DefinitionBody":
        "openapi": "3.0.2"
        "paths":
          "/myOperation":
              "id-format": "$arn{arn:${stageVariables.arnPartition}:service::${context.identity.accountId}:resource/${method.request.body.name}}"
    "Type": "AWS::Serverless::Api"
```

**Comment by kddejong** [comment]
I haven't been able to replicate this issue with the template provided.  Since it is getting sent through SAM I want to see the completed Body for the openapi schema.

**Comment by cgledezma1101** [comment]
I did a bit more investigation on this, and it seems it might be related to [this issue](https://github.com/aws/serverless-application-model/issues/1590), and not the actual regex. Feel free to validate and close

**Comment by PatMyron** [comment]
SAM issue covered in https://github.com/aws-cloudformation/cfn-lint/pull/2513
